component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/voiceotp/VoiceOTPAuthenticator.java

```diff
@@ -0,0 +1,2037 @@
+     */
+    private long getUnlockTimeInMilliSeconds(AuthenticatedUser authenticatedUser) throws AuthenticationFailedException {
+
+        String username = authenticatedUser.toFullQualifiedUsername();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        try {
+            UserRealm userRealm = getUserRealm(username);
+            if (userRealm == null) {
+                throw new AuthenticationFailedException("UserRealm is null for user : " + username);
+            }
+            UserStoreManager userStoreManager = userRealm.getUserStoreManager();
+            if (userStoreManager == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("userStoreManager is null for user: " + username);
+                }
+                throw new AuthenticationFailedException("userStoreManager is null for user: " + username);
+            }
+            Map<String, String> claimValues = userStoreManager
+                    .getUserClaimValues(tenantAwareUsername, new String[]{VoiceOTPConstants.ACCOUNT_UNLOCK_TIME_CLAIM},
+                            null);
+            if (claimValues.get(VoiceOTPConstants.ACCOUNT_UNLOCK_TIME_CLAIM) == null) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("No value configured for claim: %s, of user: %s",
+                            VoiceOTPConstants.ACCOUNT_UNLOCK_TIME_CLAIM, username));
+                }
+                return 0;
+            }
+            return Long.parseLong(claimValues.get(VoiceOTPConstants.ACCOUNT_UNLOCK_TIME_CLAIM));
+        } catch (UserStoreException e) {
+            throw new AuthenticationFailedException("Cannot find the user claim for unlock time for user : " +
+                    username, e);
+        }
+    }
+
+    /**
+     * This method returns the boolean value of the mobile number update failed context property.
+     *
+     * @param context
+     * @return The status of mobile number update failed parameter
+     */
+    private boolean isMobileNumberUpdateFailed(AuthenticationContext context) {
+
+        return Boolean.parseBoolean
+                (String.valueOf(context.getProperty(VoiceOTPConstants.MOBILE_NUMBER_UPDATE_FAILURE)));
+    }
+
+    /**
+     * This method returns the boolean value of the code mismatch context property.
+     *
+     * @param context
+     * @return The value of the code mismatch parameter
+     */
+    private boolean isCodeMismatch(AuthenticationContext context) {
+
+        return Boolean.parseBoolean(String.valueOf(context.getProperty(VoiceOTPConstants.CODE_MISMATCH)));
+    }
+
+    /**
+     * Split the given number by provided divisor.
+     *
+     * @param context Authentication context.
+     * @return split number
+     */
+    private String splitAndEncodeNumber(String otp, int divisor, AuthenticationContext context) {
```

Comment:
Is there any encoding happening here? If not shall we use a more meaningful method name like `formatOTP`? and change the method description to emphasize this behaviour?